src/plugins/data/public/search/expressions/es_raw_response.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { SearchResponse } from 'elasticsearch';
+
+const name = 'es_raw_response';
+
+export interface EsRawResponse {
+  type: typeof name;
+  body: SearchResponse<unknown>;
+}
+
+function flatten(obj: any, keyPrefix = '') {
```

Comment:
I'm pretty sure we've already got this same sort of code several times throughout our code base (see https://github.com/elastic/kibana/blob/41f085adccd54b5d3673777ab829651726e00f4c/x-pack/plugins/snapshot_restore/common/lib/flatten.ts). Is this doing something special for arrays or would it be possible to re-use code from somewhere else?

Reply:
this is still work in progress, but i think this will be very specific to es response, so i am not sure if we can reuse something else. this at the moment only works with aggregations, need to add support for raw docs and probably handle some of the edge cases as well.